### ISSUE TITLE
[raw2science] Performance issues for large alert DB

### ISSUE THREAD

**Issue by JulienPeloton** [astrolabsoftware/fink-broker#336]
**Describe the issue**

The first step of `raw2science` is to connect to the raw Parquet database, and start streaming new data:

```python
# in sparkUtils.connect_to_raw_database
spark = SparkSession \
    .builder \
    .getOrCreate()

# Create a DF from the database
userschema = spark\
    .read\
    .parquet(basepath)\
    .schema

df = spark \
    .readStream \
    .format("parquet") \
    .schema(userschema) \
    .option("basePath", basepath) \
    .option("path", path) \
    .option("latestFirst", latestfirst) \
    .load()
```

We carefully chose `latestfirst=True` such that we process the latest new files first (useful when there is a large backlog of files). But despite that, for large volume of alerts (say > 10 full days, or 100 GB of alerts), the job is hanging - sometimes for hours! - before starting streaming and processing the data. The log is pretty telling (look at the times!):

```
20/03/20 07:22:48 INFO handler.ContextHandler: Started o.s.j.s.ServletContextHandler@123d2317{/SQL/execution/json,null,AVAILABLE,@Spark}
20/03/20 07:22:48 INFO handler.ContextHandler: Started o.s.j.s.ServletContextHandler@642f4910{/static/sql,null,AVAILABLE,@Spark}
20/03/20 07:22:48 INFO state.StateStoreCoordinatorRef: Registered StateStoreCoordinator endpoint
20/03/20 08:41:02 WARN datasources.SharedInMemoryCache: Evicting cached table partition metadata from memory due to size constraints (spark.sql.hive.filesourcePartitionFileCacheSize = 262144000 bytes). This may impact query planning performance.
20/03/20 08:41:03 INFO main (raw2science.py line 62): fink_broker.filters.qualitycuts
20/03/20 08:41:03 INFO apply_user_defined_filter (filters.py line 239): new filter/topic registered: qualitycuts from fink_broker.filters
20/03/20 08:41:04 INFO main (raw2science.py line 66): New processor: cdsxmatch
20/03/20 08:41:04 INFO main (raw2science.py line 75): New processor: rfscore
20/03/20 08:41:04 WARN streaming.StreamingQueryManager: spark.sql.adaptive.enabled is not supported in streaming DataFrames/Datasets and will be disabled.
20/03/20 08:41:05 WARN streaming.FileStreamSource: 'latestFirst' is true. New files will be processed first, which may affect the watermark
value. In addition, 'maxFileAge' will be ignored.
20/03/20 10:04:13 WARN streaming.FileStreamSource: Listed 188006 file(s) in 4978403.839649 ms
```

The first `WARN` is telling that there is a problem with data volume - but that's explicitly the reason why we put `latestfirst=True`! And I would expect the third  `WARN` to appear right away - not 1.30h after the launch of the job! This could be a sign of a bad cluster configuration or bad data partitioning (e.g. too many small partitions).

Note the internet is silent about a possible reason and a possible solution - so I might do something dumb and it must be trivial to solve...

Here are a couple of things to test to reduce the latencies:
- increase the trigger when we write stream (`stream2raw`), so that more data falls into the same file. It is currently in `as quick as possible` mode.
- Modify the partitioning of the data. It is currently partitioned by `"year", "month", "day", "hour"` - and maybe dropping the `hour` would help?
- Check the config script of the cluster - especially everything related to hive partitioning. 

Of course, increasing the resources in the cluster might help - but that's not the kind of behaviour we expect from Spark!

**Comment by JulienPeloton** [comment]
Superseded by  #340.
